# INFERENCIA ESTADISTICA: Quinta practica

## Analisis de la varianza

Vamos a considerar la base de datos del fichero excel alturas_AOV.xlsx (es la misma base de datos que en las anteriores practicas, pero con una variable adicional con las alturas de los hijos. Esa variable se llamará HIJO)

Nuestro objetivo es comparar si hay diferencia entre las medias de la altura de los hijos entre los que son de Alava, Vizcaya y Guipuzcoa usando el analisis de la varianza.

In [92]:
# leer fichero xls
library("readxl")

Datos <-read_excel("alturas_AOV.xlsx")

str(Datos)

# vemos que PROVINCIA es una variable entera. Como para el analisis de la varianza, necesitamos que esa variable
# sea cualitativa (factor), creamos PROVINCIA.F como lo hicimos en la primera practica

Datos$PROVINCIA.F <- factor(Datos$PROVINCIA, levels=c(1,2,3), 
                              labels=c("Alava", "Vizcaya", "Guipuzcoa"))


tibble [171 × 8] (S3: tbl_df/tbl/data.frame)
 $ ID       : num [1:171] 1 2 3 4 5 6 7 8 9 10 ...
 $ PADRE    : num [1:171] 174 177 173 174 160 167 171 174 175 174 ...
 $ MADRE    : num [1:171] 156 159 161 156 165 157 162 158 162 161 ...
 $ SEXO     : chr [1:171] "mujer" "hombre" "hombre" "hombre" ...
 $ PROVINCIA: num [1:171] 2 2 1 2 1 3 1 3 3 1 ...
 $ ALTURA   : num [1:171] 165 170 168 167 162 163 165 168 168 167 ...
 $ PESO     : num [1:171] 65 67 51 69 54 61 65 76 67 69 ...
 $ HIJO     : num [1:171] 165 170 153 167 147 163 150 168 168 152 ...


In [93]:
# vemos que PROVINCIA.F es tipo factor
str(Datos)

tibble [171 × 9] (S3: tbl_df/tbl/data.frame)
 $ ID         : num [1:171] 1 2 3 4 5 6 7 8 9 10 ...
 $ PADRE      : num [1:171] 174 177 173 174 160 167 171 174 175 174 ...
 $ MADRE      : num [1:171] 156 159 161 156 165 157 162 158 162 161 ...
 $ SEXO       : chr [1:171] "mujer" "hombre" "hombre" "hombre" ...
 $ PROVINCIA  : num [1:171] 2 2 1 2 1 3 1 3 3 1 ...
 $ ALTURA     : num [1:171] 165 170 168 167 162 163 165 168 168 167 ...
 $ PESO       : num [1:171] 65 67 51 69 54 61 65 76 67 69 ...
 $ HIJO       : num [1:171] 165 170 153 167 147 163 150 168 168 152 ...
 $ PROVINCIA.F: Factor w/ 3 levels "Alava","Vizcaya",..: 2 2 1 2 1 3 1 3 3 1 ...


### Contrastamos la normalidad de las alturas de Alava, Vizcaya y Guipuzcoa

Consideramos un nivel de confianza del 5%

In [94]:
# Contrastamos la normalidad de las alturas de los hijos de los de Alava con nivel de significación 5%
shapiro.test(Datos$HIJO[Datos$PROVINCIA.F=="Alava"])
# el p-valor es mayor que 0.05, no podemos rechazar la normalidad de las alturas s de los hijos de los de Alava


	Shapiro-Wilk normality test

data:  Datos$HIJO[Datos$PROVINCIA.F == "Alava"]
W = 0.95762, p-value = 0.1485


In [95]:
# Contrastamos la normalidad de las alturas de los hijos de los de Vizcaya con nivel de significación 5%
shapiro.test(Datos$HIJO[Datos$PROVINCIA.F=="Vizcaya"])
# el p-valor es mayor que 0.05 => no podemos rechazar la normalidad de las alturas de los hijos de los de Vizcaya


	Shapiro-Wilk normality test

data:  Datos$HIJO[Datos$PROVINCIA.F == "Vizcaya"]
W = 0.98345, p-value = 0.373


In [96]:
# Contrastamos la normalidad de las alturas de los hijos de los de Guipuzcoa con nivel de significación 5%
shapiro.test(Datos$HIJO[Datos$PROVINCIA.F=="Guipuzcoa"])
# el p-valor es mayor que 0.05 => no podemos rechazar la normalidad de los hijos de las alturas de los de Guipuzcoa


	Shapiro-Wilk normality test

data:  Datos$HIJO[Datos$PROVINCIA.F == "Guipuzcoa"]
W = 0.98188, p-value = 0.6335


### Contrastamos la igualdad de la varianza de las alturas de Alava, Vizcaya y Guipuzcoa

Consideramos un nivel del confianza del 2%

In [97]:
# Usaremos el comando leveneTest. Hay que cargar (e instalarlo si no lo habeis usado antes) el paquete car
#install.packages("car")
#library("car")
leveneTest(Datos$HIJO, Datos$PROVINCIA.F, center=mean) 

Df  F value  Pr(>F)   
group   2 1.005729 0.3679694
      168       NA        NA

#### Explicación:
El valor del estadistico del test de Levene es 1.005729, los grados de libertad 2 y el p-valor es 0.3679694

#### Conclusión:
Como el p-valor es mayor que 0.02, no podemos rechazar la igualdad de varianzas de las alturas de los de Alava, Vizcaya y Guipuzcoa

### Analisis de la varianza

Ninguno de los test anteriores, nos ha afirmado que no se cumple alguna de las hipotesis para realizar el analisi de la varianza. Por tanto, todo parece indicar que estamos bajo las hipotesis para hacer este analisis.

Consideramos un nivel de confianza del 5%.

In [102]:
Anova <-  aov(HIJO ~ PROVINCIA.F,data=Datos)
summary(Anova)

             Df Sum Sq Mean Sq F value Pr(>F)    
PROVINCIA.F   2   6705    3353     299 <2e-16 ***
Residuals   168   1884      11                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#### Explicacion:
El valor del estadistico de contraste es 299 y el p-valor es menor a 0.0001. Por tanto, como el p-valor es inferior a 0.05, concluismo que hay diferencia en las alturas de los hijos segun la provincia de donde son.

### Comparaciones multiples

Sabemos que no son iguales la media de las alturas de los hijos de los de Alava, Vizcaya y Guipuzcoa. Ahora, queremos ir más allá y saber más sobre esta diferencia. Usaremos comparaciones multiples. Hay distintos metodos para hacer estas comparaciones multiples, pero los más usados son el de Scheffe y el de Tukey. Para el de Scheffe, se require que el numero de muestras en cada grupo sea el mismo. Por eso, usaremos el de Tukey en general. 

Vamos a considerar un nivel de significación del 2%.

In [108]:
# cargar el siguiente paquete (e instalarlo si lo vais a usar por primera vez)
library(multcomp)
comp_Tukey <- glht(Anova, linfct = mcp(PROVINCIA.F = "Tukey"))
summary(comp_Tukey)


	 Simultaneous Tests for General Linear Hypotheses

Multiple Comparisons of Means: Tukey Contrasts


Fit: aov(formula = HIJO ~ PROVINCIA.F, data = Datos)

Linear Hypotheses:
                         Estimate Std. Error t value Pr(>|t|)    
Vizcaya - Alava == 0     14.94528    0.65134  22.945   <1e-06 ***
Guipuzcoa - Alava == 0   14.88821    0.71537  20.812   <1e-06 ***
Guipuzcoa - Vizcaya == 0 -0.05707    0.60083  -0.095    0.995    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
(Adjusted p values reported -- single-step method)


#### Explicación:
Hay tres contrastes de hipotesis:
* El primer contraste de hipotesis contrasta la diferencia entre la media de las alturas de Vizcaya ($\mu_V$) y de Alava ($\mu_A$), es decir, 

$$H_0: \mu_{V}  =  \mu_{A}$$
$$H_1: \mu_{V}\neq \mu_{A}$$

Para este contraste, se ve que el valor del estadistico es 22.945 y el p-valor es inferior a 0.0001. Por tanto, como el p-valor es inferior a 0.02, concluimos que la media de la altura de los hijos de los de Alava y los Vizcaya es distinta.

* El segundo contraste de hipotesis contrasta la diferencia entre la media de las alturas de Guipuzcoa ($\mu_G$) y de Alava, es decir, 

$$H_0: \mu_{G}  =  \mu_{A}$$
$$H_1: \mu_{G}\neq \mu_{A}$$

Para este contraste, se ve que el valor del estadistico es 20.812 y el p-valor es inferior a 0.0001. Por tanto, como el p-valor es inferior a 0.02, concluimos que la media de la altura de los hijos de los de Alava y los Guipuzcoa es distinta.

* El tercer contraste de hipotesis contrasta la diferencia entre la media de las alturas de Guipuzcoa y los de Vizcaya, es decir, 

$$H_0: \mu_{G}  =  \mu_{V}$$
$$H_1: \mu_{G}\neq \mu_{V}$$

Para este contraste, se ve que el valor del estadistico es -0.095 y el p-valor es inferior a  0.995. Por tanto, como el p-valor es superior a 0.02, no podemos rechazar la igualdad de medias de la altura de los hijos de los de Guipuzcoa y los Vizcaya.

#### Conclusion:

La media de la altura de los hijos de Alava es distinta a la media de la altura de los hijos de los de Guipuzcoa y de los de Vizcaya. Sin embargo, no podemos concluir que haya diferencia entre la altura media de los hijos de los de Vizcaya y Guipuzcoa.


## Intervalos de confianza

Sabemos que la altura media de los hijos de Alava es distinta a la altura media de los hijos de los de Guipuzcoa y de los de Vizcaya. Para saber si es mayor o menor, usamos intervalos de confianza.


In [109]:
confint(comp_Tukey)


	 Simultaneous Confidence Intervals

Multiple Comparisons of Means: Tukey Contrasts


Fit: aov(formula = HIJO ~ PROVINCIA.F, data = Datos)

Quantile = 2.3611
95% family-wise confidence level
 

Linear Hypotheses:
                         Estimate lwr      upr     
Vizcaya - Alava == 0     14.94528 13.40743 16.48313
Guipuzcoa - Alava == 0   14.88821 13.19917 16.57724
Guipuzcoa - Vizcaya == 0 -0.05707 -1.47566  1.36151


#### Explicación:

Hemos calculado tres intervalos de confianza:
$$I_{\mu_V-\mu_A}^{0.95}=(13.40743,16.48313)$$
$$I_{\mu_G-\mu_A}^{0.95}=(13.19917,16.57724)$$
$$I_{\mu_G-\mu_A}^{0.95}=(-1.47566,1.36151)$$

Del primer intervalo de confianza concluimos que $$\mu_A<\mu_V$$ y del segundo que $$\mu_A<\mu_G.$$ Por tanto, la altura media de los hijos de los de Alava es inferior a la altura media de los hijos de los de Vizcaya y de los hijos de los de Guipuzcoa.